### PR TITLE
Test flake intersection observer

### LINKED ISSUE #5359: IntersectionObserver test fails consistently on master
See https://travis-ci.org/ampproject/amphtml/builds/164163236

```
ium 51.0.2704 (Ubuntu 0.0.0): Executed 728 of 3060 (skipped 1) SUCCESS (0 secs / 5.397 secs)
Chromium 51.0.2704 (Ubuntu 0.0.0) IntersectionObserver "after each" hook for "should send more intersections" FAILED
	Error: Blocked a frame with origin "/tmp/e69ed409a334df50ee050c1b1c503acd.browserify:60810:28 <- node_modules/sinon/lib/sinon/util/core.js:154:0)
	    at each (/tmp/e69ed409a334df50ee050c1b1c503acd.browserify:58309:33 <- node_modules/sinon/lib/sinon/collection.js:34:0)
	    at Object.restore (/tmp/e69ed409a334df50ee050c1b1c503acd.browserify:58329:17 <- node_modules/sinon/lib/sinon/collection.js:54:0)
	    at Object.restore (/tmp/e69ed409a334df50ee050c1b1c503acd.browserify:59597:42 <- node_modules/sinon/lib/sinon/sandbox.js:96:0)
	    at Context.<anonymous> (/tmp/e69ed409a334df50ee050c1b1c503acd.browserify:108772:13 <- /home/travis/build/ampproject/amphtml/test/functional/test-intersection-observer.js:401:12)
```

### PR DESCRIPTION
**Pull request by jridgewell**
Stubbing on a cross-domain iframe causes errors, but we only notice them if the tests are running slowly.

The fix is to directly modify the iframe's window, not stub. Stubs are unnecessary anyways, since the iframe is in a different `window`.

Fixes #5359.

### CODE REVIEW

**Comment by camelburrito** [comment]
LGTM - would be good to explain why this happens

> Stubbing on a cross-domain iframe causes errors

File: test/functional/test-intersection-observer.js [hunk 1]
```diff
@@ -416,10 +416,10 @@ describe('IntersectionObserver', () => {
     const messages = [];
     const ioInstance = new IntersectionObserver(element, testIframe);
     insert(testIframe);
-    sandbox.stub(testIframe.contentWindow, 'postMessage', message => {
+    testIframe.contentWindow.postMessage = message => {
```
Comment (mkhatib) [inline test/functional/test-intersection-observer.js]: FMI: Why stubbing is a cause of problems here?

**Comment by jridgewell** [comment]
Apparently `Object.defineProperty` ([used by sinon](https://github.com/sinonjs/sinon/blob/master/lib/sinon/util/core/wrap-method.js#L126) during restores) will try to access privileged information in the cross-origin iframe. 💣

**Comment by mkhatib** [comment]
Cool. Can we have this comment in the presubmit explaination?

**Comment by jridgewell** [comment]
That's why there's an issue linked. 😛

**Pull request merged**